code/modules/client/preferences/entries/character/quirks.dm

```diff
@@ -0,0 +1,130 @@
+/datum/preference/choiced/quirk
+	category = PREFERENCE_CATEGORY_NON_CONTEXTUAL
+	preference_type = PREFERENCE_CHARACTER
+	can_randomize = TRUE
+	abstract_type = /datum/preference/choiced/quirk
+	/// name of the quirk to be checked for (caps sensitive)
+	var/required_quirk_name
+
+/datum/preference/choiced/quirk/apply_to_human(mob/living/carbon/human/target, value)
+	return
+
+/datum/preference/choiced/quirk/is_accessible(datum/preferences/preferences, ignore_page = FALSE)
+	if (!..())
+		return FALSE
+	if(required_quirk_name in preferences.all_quirks)
+		return TRUE
+	return FALSE
+
+/datum/preference/choiced/quirk/prosthetic_limb_location
+	required_quirk_name = "Prosthetic Limb"
+	db_key = "quirk_prosthetic_limb_location"
+
+/datum/preference/choiced/quirk/prosthetic_limb_location/init_possible_values()
+	return list(BODY_ZONE_L_ARM, BODY_ZONE_R_ARM, BODY_ZONE_L_LEG, BODY_ZONE_R_LEG)
```

Comment:
Like I said with the old PR, if you want the ability to pick a 'random' option, you will need to actually make it one of the choices. Relying on `can_randomize` doesn't really work since you can't actually make something randomize by default and it still relies on them having randomization enabled in the first place (the new randomizer is kinda dog)

Reply:
Honestly I tried to think of a way to do this, but couldn't think of a nice way at the moment, I was just gonna ask you for the best way to do it.

Reply:
Literally disable can_randomize and add another option named Random that will trigger the quirk code to just pick one. Make Random the default value.

Reply:
I tried that but it seemed to fuck up the dropdown. I'll look into it later when I feel less crap.

Reply:
Because you need to also add it to CHOICED_PREFERENCE_DISPLAY_NAMES, below.